### ISSUE TITLE
Closing echo test waiting window  while selecting the microphone joins in listen-only

### ISSUE THREAD

**Issue by GhaziTriki** [bigbluebutton/bigbluebutton#9954]
**Describe the bug**
If the echo test waiting window is closed before selecting the microphone, the conference can be joined in listen-only mode.

**To Reproduce**
Steps to reproduce the behaviour:
1. Click on join audio conference.
2. Click outside the wait window.
3. Select a microphone in the browser and accept.
4. Wait until you hear the echo test.
5. Click on the microphone icon.

**Expected behavior**
The wait window should be aware of what actions is made by the user before closing it.

**Actual behavior**
Describe in the issue.

**Screenshots**
![Rk1bsooVr8](https://user-images.githubusercontent.com/4991088/85951664-5cef9380-b96d-11ea-9946-58f9e048b4ab.gif)

**BBB version (optional):**
BigBlueButton Server 2.2.19

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro
 - Browser: Firefox Nigthly
 - Version: 79.0a1 (2020-06-27) (64 bits)

**Comment by Jossef-767** [comment]
I allwas  thought that join in listen-only mode should be the default for all cases except when choosing to connect the mic.
dont see a case where user will willingly prefer to not hear the meeting. 
the curent voice connecting proses is creating unnessery "poing of faileor" to users.
In my opinion it will be simplere to just ask "do you want to connect mic" and if the reply is "No" ,  automatecly join in listen-only mode.

**Comment by basisbit** [comment]
> dont see a case where user will willingly prefer to not hear the meeting.

Typical use case 1: user connects using a smartphone for audio + microphone and then also connects using a desktop computer for screen share / whiteboard /...

Typical use case 2: users use Discord / Mumble / phone-call for audio because they were already using that before, and then only joined the bbb session for video / whiteboard / screen sharing / ... needs

**Comment by Jossef-767** [comment]
That's true, but, most BBB users are usually first time users, in my humble opinion, it would be more natural for these users to mute audio using the volume control button on the operating system , or physical button on the speaker or headset.
In any case, one way to reduce "points of failure" for users would be to prevent option of closing the connect audio window, without first choosing MIC or listening only.

**Comment by basisbit** [comment]
In that case, could the ice candidates gathering on the client side already start directly after oage load, so that users have to wait less time for it to finish? (+ cache it as it is done twice for IPv4 only setups)

**Comment by prlanzarin** [comment]
What do you mean in "it is done twice"? There's no duplicate candidate gathering for a single SIP session. Are you talking about the IPv6 to IPv4 fallback?

**Comment by basisbit** [comment]
Go to test.bigbluebutton.org in Chrome/Chromium, join a session, open webbrowser dev tools, clean console log, select microphone connection method, wait for the echo twst to show up, in the log count how often ´RTCIceChecking Timeout Triggered after 5000 milliseconds´ shows up in the log: twice

**Comment by prlanzarin** [comment]
That doesn't mean gathering is enacted twice. It's done only once. SIP.js has two racy triggers to start the ICE checking timer which are the first ICE candidate AND ICE connection state transitions. First one wins. The checking timer is only blocking in the offer step of the negotiation (which it uses to release the SDP on the onSetLocalDescription deferred promise).

There are two instances in the log because that code is not good (even if it works). The ICE connection state transitions when the remote offer is processed, as always, and the timer is set again because it was already enacted in the offer step. tl;dr the second occurrence is a no-op, even if it's logged.

**Comment by Jossef-767** [comment]
A quick microphone connection is important.
But before the connection, some new users see a pop-up screen on the page and automatically click on the X. The result is that they remain without a voice connection.
I think the "login" process should be as foolproof as possible.

**Comment by prlanzarin** [comment]
@Jossef-767. Of course. But, please, open a new issue detailing your suggestions clearly to avoid polluting this issue (which is a legitimate bug report). This has to be fixed regardless of which medium term changes we need to do to improve the audio experience.

**Comment by prlanzarin** [comment]
Can't reproduce as of 2.3.
